### ISSUE TITLE
offline mode pwa

### ISSUE THREAD

**Issue by sbounmy** [sbounmy/hongbao_bitcoin#18]
so we can generate paper wallet without having internet

current :
we generate the bitcoin keys using @bitcoin-ruby gem (our own fork https://github.com/sbounmy/bitcoin-ruby/tree/bip39-mnemonic)
server side require trust so its better for audit and verification to do it on front end only.

- move keys generation (mnemonic / private / pub) on front end 
- move qrcode creation to front end
- check if assets (bill images) works correctly on offline (Save page As...)

Ofc top up a bill with Mt Pelerin will not be possible without internet

**Comment by sbounmy** [comment]
its working but facing issue when `Save page As`...
![CleanShot 2024-12-19 at 22 09 04 2](https://github.com/user-attachments/assets/f06d73ea-350c-4e14-9ede-d69de016a9b9)

https://stackoverflow.com/questions/79297035/importmap-and-save-page-as-in-browser-does-not-save-js-files

**Comment by SalmaTalbi** [comment]
Hello @sbounmy is the "save as issue" still open?

**Comment by sbounmy** [comment]
Yes close this because its works offline (disconnect internet) but when saving page as does not save asset . it's a separate ticket #91
